### ISSUE TITLE
Array Iteration (from Issue 2)

### ISSUE THREAD

**Issue by anywhichway** [anywhichway/rule-reactor#4]
Is it possible for a Patient could have an array of addresses and a rule that iterates over the addresses and fires if the doctor's address is in the same city as one of them?

**Comment by anywhichway** [comment]
Yes, just pass-in the object that has a property with an array value and iterate over the array using the Javascript .some function, e.g.

```
function(p1,p2) {
   return p1.locations.some((location) => { return location.city==p2.address.city; }
}
```

Although some Rete matching rule engines have the ability to match sequence offsets using a special operator, rule-reactor does not support this because is does not have a full pattern and join network.

**Comment by tve** [comment]
Aha, thanks for the response. I think the real question I had was whether the object in the array gets the hooks that triggers rule evaluation. E.g., if `p1.locations[3].City` gets modified, does this retrigger evaluation of the appropriate rules (just like a modification of a simple property of a domain object does). Or am I completely misunderstanding things?

**Comment by anywhichway** [comment]
Your use case is not supported at this time, it matches what I described at the end of my last comment. Arrays are tricky things to modify and observe given the removal of Object.observe. We could do this with a Proxy, but it was not available and stable at the time we wrapped initial development. We may add it in the future, but I can't give you a timeframe.

A workaround would be to add a special property to the object that you update any time you change the array. This will force rules to re-evaluate so that the loop over the array is tested again.
